hibernate-cache/src/main/java/org/hibernate/cache/infinispan/access/TombstoneCallInterceptor.java

```diff
@@ -169,22 +168,23 @@ private Object setValue(MVCCEntry e, Object value) {
 			e.setChanged(true);
 		}
 		if (value instanceof Tombstone) {
+         command.setMetadata( expiringMetadata );
 			e.setMetadata(expiringMetadata);
 		}
 		else {
+         command.setMetadata( defaultMetadata );
 			e.setMetadata(defaultMetadata);
 		}
 		return e.setValue(value);
 	}
 
-	private void setFailed(PutKeyValueCommand command) {
+	private Object setFailed(InvocationContext ctx, PutKeyValueCommand command) {
 		// This sets command to be unsuccessful, since we don't want to replicate it to backup owner
 		command.setValueMatcher(ValueMatcher.MATCH_NEVER);
-		try {
-			command.perform(null);
-		}
-		catch (Throwable ignored) {
-		}
+      return invokeNextAndExceptionally( ctx, command, (rCtx, rCommand, throwable) -> {
```

Comment:
Therefore you should remove method `setFailed` completely and just call `command.fail()` on the places where you call it.